lib/libstatcounters/statcounters_mips.inc

```diff
@@ -35,62 +38,90 @@
-STATCOUNTER_ITEM(inst_user,4,1)
-STATCOUNTER_ITEM(inst_kernel,4,2)
-STATCOUNTER_ITEM(imprecise_setbounds,4,3)
-STATCOUNTER_ITEM(unrepresentable_caps,4,4)
-STATCOUNTER_ITEM(itlb_miss,5,0)
-STATCOUNTER_ITEM(dtlb_miss,6,0)
-STATCOUNTER_ITEM(icache_write_hit,8,0)
-STATCOUNTER_ITEM(icache_write_miss,8,1)
-STATCOUNTER_ITEM(icache_read_hit,8,2)
-STATCOUNTER_ITEM(icache_read_miss,8,3)
-STATCOUNTER_ITEM(icache_evict,8,6)
-STATCOUNTER_ITEM(dcache_write_hit,9,0)
-STATCOUNTER_ITEM(dcache_write_miss,9,1)
-STATCOUNTER_ITEM(dcache_read_hit,9,2)
-STATCOUNTER_ITEM(dcache_read_miss,9,3)
-STATCOUNTER_ITEM(dcache_evict,9,6)
-STATCOUNTER_ITEM(dcache_set_tag_write,9,8)
-STATCOUNTER_ITEM(dcache_set_tag_read,9,9)
-STATCOUNTER_ITEM(l2cache_write_hit,10,0)
-STATCOUNTER_ITEM(l2cache_write_miss,10,1)
-STATCOUNTER_ITEM(l2cache_read_hit,10,2)
-STATCOUNTER_ITEM(l2cache_read_miss,10,3)
-STATCOUNTER_ITEM(l2cache_evict,10,6)
-STATCOUNTER_ITEM(l2cache_set_tag_write,10,8)
-STATCOUNTER_ITEM(l2cache_set_tag_read,10,9)
-STATCOUNTER_ITEM(mem_byte_read,11,0)
-STATCOUNTER_ITEM(mem_byte_write,11,1)
-STATCOUNTER_ITEM(mem_hword_read,11,2)
-STATCOUNTER_ITEM(mem_hword_write,11,3)
-STATCOUNTER_ITEM(mem_word_read,11,4)
-STATCOUNTER_ITEM(mem_word_write,11,5)
-STATCOUNTER_ITEM(mem_dword_read,11,6)
-STATCOUNTER_ITEM(mem_dword_write,11,7)
-STATCOUNTER_ITEM(mem_cap_read,11,8)
-STATCOUNTER_ITEM(mem_cap_write,11,9)
-STATCOUNTER_ITEM(mem_cap_read_tag_set,11,10)
-STATCOUNTER_ITEM(mem_cap_write_tag_set,11,11)
-STATCOUNTER_ITEM(tagcache_write_hit,12,0)
-STATCOUNTER_ITEM(tagcache_write_miss,12,1)
-STATCOUNTER_ITEM(tagcache_read_hit,12,2)
-STATCOUNTER_ITEM(tagcache_read_miss,12,3)
-STATCOUNTER_ITEM(tagcache_evict,12,6)
-STATCOUNTER_ITEM(l2cachemaster_read_req,13,0)
-STATCOUNTER_ITEM(l2cachemaster_write_req,13,1)
-STATCOUNTER_ITEM(l2cachemaster_write_req_flit,13,2)
-STATCOUNTER_ITEM(l2cachemaster_read_rsp,13,3)
-STATCOUNTER_ITEM(l2cachemaster_read_rsp_flit,13,4)
-STATCOUNTER_ITEM(l2cachemaster_write_rsp,13,5)
-STATCOUNTER_ITEM(tagcachemaster_read_req,14,0)
-STATCOUNTER_ITEM(tagcachemaster_write_req,14,1)
-STATCOUNTER_ITEM(tagcachemaster_write_req_flit,14,2)
-STATCOUNTER_ITEM(tagcachemaster_read_rsp,14,3)
-STATCOUNTER_ITEM(tagcachemaster_read_rsp_flit,14,4)
-STATCOUNTER_ITEM(tagcachemaster_write_rsp,14,5)
+#ifndef STATCOUNTERS_GROUP_END
+#define STATCOUNTERS_GROUP_END()
+#endif
+
+/*
+ * This list is derived from the BERI Hardware reference (SVN snapshot)
+ * Table 8.4: "List of available STATCOUNTERS counters".
+ * Note: Not included in the latest released version (UCAM-CL-TR-868.pdf).
+ * The counters inst_user,inst_kernel,imprecise_setbounds,unrepresentable_caps
+ * are not listed in that table but are available in CHERI-MIPS QEMU.
+ */
+STATCOUNTER_ITEM(cycles, cycles, (2, 0))
+STATCOUNTER_ITEM(instructions, instructions, (4, 0))
+STATCOUNTER_ITEM(instructions(user), inst_user, (4, 1))
+STATCOUNTER_ITEM(instructions(kernel), inst_kernel, (4, 2))
+MIPS_STATCOUNTER(itlb_miss, 5, 0)
```

Comment:
Do we ever care about the distinction anywhere?

Reply:
I think we almost always summed both of these for plots, but I believe the BERI statcounters report them both separately so we might as well have both. I don't want to add special-cased code to sum them in libstatcounters. Also MIPS will go away so this is a waste of effort.